controllers/user.controller.js

```diff
@@ -1,32 +1,40 @@
+const bcrypt = require('bcryptjs');
+
 const User = require('../models/user.model');
 
 // eslint-disable-next-line consistent-return
 const addUser = async (req, res) => {
-  const user = req.body;
-  const { email } = req.body;
-  user.posts = [];
+  const { username, email, password } = req.body;
 
   try {
-    const userExists = await User.findOne({ email });
+    const user = await User.findOne({ email });
 
-    if (userExists) {
+    if (user) {
       return res.status(400).json({
         ok: false,
-        msg: 'Ya existe un usuario con ese correo'
+        msg: 'Ya existe un usuario con ese correo',
       });
     }
-    const newUser = new User(user);
+    const userContent = { username, email, password };
+    userContent.posts = [];
+    userContent.comments = [];
+
+    const newUser = new User(userContent);
```

Comment:
Check suggestion works, so you can take advantage of destructuring
`const newUser = new User({
  username,
  email,
  password
});`